### PR TITLE
[metal3] add data-test-id for dashboard components

### PR DESCRIPTION
**Pull request by dantrainor**

### CODE REVIEW

**Comment by dantrainor** [comment]
Amended the commit to remove frontend/public/components/dashboard/inventory-card/inventory-item.tsx, it was erroneously included; it is instead part of #2272

File: frontend/packages/metal3-plugin/src/components/dashboard/inventory-card.tsx [hunk 1]
```diff
@@ -94,6 +94,7 @@ export const InventoryCard: React.FC<InventoryCardProps> = ({
           pluralTitle="Pods"
           count={podCount}
           error={podError}
+          data-test-id="metal3-plugin-dashboard-pod"
```
Comment (jtomasek) [inline frontend/packages/metal3-plugin/src/components/dashboard/inventory-card.tsx]: Is the naming based on some convention used in console? I think current name could easily become duplicate if we have several things related to pods in the dashboard.

how about `metal3-plugin-dashboard-inventory-pods` ?

**Comment by spadgett** [comment]
Please add the attributes in the same PR as the tests so we can see how they will be used.

**Comment by dantrainor** [comment]
> Please add the attributes in the same PR as the tests so we can see how they will be used.

Just to clarify, are you suggesting that #1886 change to use the attributes named in this PR, or merge these changes (and the ones from #2287) into #1886 ?

**Comment by spadgett** [comment]
> > Please add the attributes in the same PR as the tests so we can see how they will be used.
> 
> Just to clarify, are you suggesting that #1886 change to use the attributes named in this PR, or merge these changes (and the ones from #2287) into #1886 ?

I'm suggesting to merge the changes so it's all one PR. That makes it easier to understand and review.

**Comment by dantrainor** [comment]
> > > Please add the attributes in the same PR as the tests so we can see how they will be used.
> > 
> > 
> > Just to clarify, are you suggesting that #1886 change to use the attributes named in this PR, or merge these changes (and the ones from #2287) into #1886 ?
> 
> I'm suggesting to merge the changes so it's all one PR. That makes it easier to understand and review.

Understood.  Per @honza 's comment on #2287 this was split to ease review burden.

**Comment by jtomasek** [comment]
+1 to adding this to the tests PR #1886 so the usage of the ids is clear

**Comment by dantrainor** [comment]
This is going to become part of #1886

**Pull request closed without merging**